src/pages/kb/open-source/dev-guide/write-a-query-runner.md

```diff
@@ -0,0 +1,268 @@
+---
+category: dev-guide
+parent_category: open-source
+title: Writing a New Query Runner 
+slug: write-a-query-runner
+toc: true
+---
+
+## Intro 
+Redash already connects to [many]({% link _kb/data-sources/querying/supported-data-sources.md %}) databases and REST APIs. To add support for a new data source type in Redash, you need to implement a Query Runner for it. A Query Runner is a Python class. This doc page shows the process of writing a new Query Runner. It uses the Firebolt Query Runner as an example.
+
+Start by creating a new `firebolt.py` file in the `/redash/query_runner` directory and implement the `BaseQueryRunner` class:
+
+```python
+from redash.query_runner import BaseQueryRunner, register
+
+class Firebolt(BaseQueryRunner):
+    def run_query(self, query, user):
+        pass
+```
+
+The only method that you must implement is the `run_query` method, which accepts a `query` parameter (string) and the `user` who invoked this query. The user is irrelevant for most query runners and can be ignored.
+
+## Configuration
```

Comment:
It's worth showing an example for:
1. Adding a friendly name and description for fields.
2. File type fields.

We should at least call out it's possible.

Reply:
Great catch. Fixed in bae95c6